pkg/master/controller.go

```diff
@@ -200,6 +197,19 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Type:            serviceType,
 		},
 	}
+	hash := strconv.FormatUint(hashService(svc), 10)
+	svc.Annotations = map[string]string{"kubernetes.io/config.hash": hash}
```

Comment:
Putting the hash in the annotations makes the hash wrong, unless you want to do something like bitcoin mining to make it not wrong ;)

Just saying...